pkg/oc/cli/admin/groups/sync/sync.go

```diff
@@ -75,147 +74,104 @@ const (
 
 var AllowedSourceTypes = []string{string(GroupSyncSourceLDAP), string(GroupSyncSourceOpenShift)}
 
-func ValidateSource(source GroupSyncSource) bool {
-	return sets.NewString(AllowedSourceTypes...).Has(string(source))
-}
-
 type SyncOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+	Printer    printers.ResourcePrinter
+
 	// Source determines the source of the list of groups to sync
 	Source GroupSyncSource
 
 	// Config is the LDAP sync config read from file
-	Config *config.LDAPSyncConfig
+	Config     *config.LDAPSyncConfig
+	ConfigFile string
 
 	// Whitelist are the names of OpenShift group or LDAP group UIDs to use for syncing
-	Whitelist []string
+	Whitelist     []string
+	WhitelistFile string
 
 	// Blacklist are the names of OpenShift group or LDAP group UIDs to exclude
-	Blacklist []string
+	Blacklist     []string
+	BlacklistFile string
+
+	Type string
 
 	// Confirm determines whether or not to write to OpenShift
 	Confirm bool
 
-	// GroupInterface is the interface used to interact with OpenShift Group objects
-	GroupInterface userv1client.GroupInterface
-
-	// Stderr is the writer to write warnings and errors to
-	Stderr io.Writer
-
-	// Out is the writer to write output to
-	Out io.Writer
-}
-
-// CreateErrorHandler creates an error handler for the LDAP sync job
-func (o *SyncOptions) CreateErrorHandler() syncerror.Handler {
-	components := []syncerror.Handler{}
-	if o.Config.RFC2307Config != nil {
-		if o.Config.RFC2307Config.TolerateMemberOutOfScopeErrors {
-			components = append(components, syncerror.NewMemberLookupOutOfBoundsSuppressor(o.Stderr))
-		}
-		if o.Config.RFC2307Config.TolerateMemberNotFoundErrors {
-			components = append(components, syncerror.NewMemberLookupMemberNotFoundSuppressor(o.Stderr))
-		}
-	}
+	// GroupClient is the interface used to interact with OpenShift Group objects
+	GroupClient userv1typedclient.GroupsGetter
 
-	return syncerror.NewCompoundHandler(components...)
+	genericclioptions.IOStreams
 }
 
-func NewSyncOptions() *SyncOptions {
+func NewSyncOptions(streams genericclioptions.IOStreams) *SyncOptions {
 	return &SyncOptions{
-		Stderr:    os.Stderr,
-		Whitelist: []string{},
+		Whitelist:  []string{},
+		Type:       string(GroupSyncSourceLDAP),
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme).WithDefaultOutput("yaml"),
```

Comment:
I think this needs a success message

Reply:
There was none before that's why I left it empty. We can always fix it later on if need to.